File: backend/models/ServiceTemplate.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/db');

const ServiceTemplate = sequelize.define('ServiceTemplate', {
  id: { type: DataTypes.INTEGER, primaryKey: true, autoIncrement: true },
  name: { type: DataTypes.STRING, allowNull: false },
  interval_days: { type: DataTypes.INTEGER },
  interval_km: { type: DataTypes.INTEGER },
  vehicle_type: { type: DataTypes.STRING },
  user_id: { type: DataTypes.INTEGER, allowNull: true } // Support multi-device isolation
}, {
  tableName: 'tbl_service_templates',
  timestamps: false
});

module.exports = ServiceTemplate;

File: mechminder/lib/models/service_template.dart
class ServiceTemplate {
  final int? id;
  final String name;
  final int? intervalDays;
  final int? intervalKm;

  ServiceTemplate({
    this.id,
    required this.name,
    this.intervalDays,
    this.intervalKm,
  });

  factory ServiceTemplate.fromMap(Map<String, dynamic> map) {
    return ServiceTemplate(
      id: map['id'] ?? map['_id'],
      name: map['name'] ?? '',
      intervalDays: map['interval_days'] ?? map['intervalDays'],
      intervalKm: map['interval_km'] ?? map['intervalKm'],
    );
  }

  Map<String, dynamic> toMap() {
    return {
      if (id != null) '_id': id,
      'name': name,
      'interval_days': intervalDays,
      'interval_km': intervalKm,
    };
  }
}
